storybookjs/storybook
# Vue3: Update args without re-mounting component

Issue: https://github.com/storybookjs/storybook/issues/13913

## What I did

Change `app:vue3` to update args, when user updates controls instead of re-mounting the entire component
- reassign `activeStoryComponent` only, when `forceRender` is not true (e.g. when changing stories)
- add `storyArgs` data to root vue component
- when rendering the actual story component, reading the `storyArgs` from the root vue component and use those as props

I based the changes on what I understand was done in `app:vue`, although everything is a bit different there. Happy to improve this further if anyone has suggestions.

## How to test

- Run example `vue-3-cli`
- Add the following to `Button.vue`:
```js
mounted() {
  console.log('mounted')
}
```
- Open storybook: http://localhost:6006/?path=/story/example-button--primary
- Update controls → there should not be any console log `mounted`
- Change story → there should be a console log `mounted`

I feel like this does not need an example, happy to include one if wanted though
